docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+---
+title: 'Advanced | Lua Authentication'
+---
+
+## Introduction
+
+Dovecot has the ability to let users create their own custom user provisioning and authentication providers in [Lua](https://en.wikipedia.org/wiki/Lua_(programming_language)#Syntax). This allows any data source that can be approached from Lua to be used for authentication, including web servers. It is possible to do more with Dovecot and Lua, but other use cases fall outside of the scope of this documentation page.
+
+!!! warning
+
+    DMS offers minimal support for Lua-based authentication due to it being an advanced method that can be used in many different ways. Do not open issues on GitHub or request support from DMS contributors for Lua scripts. The documentation on this page is all that is needed to get your own development started. Look elsewhere for Lua support.
+
+!!! warning
+
+    Lua-based authentication relies on a Dovecot plugin. Dovecot is known to sometimes deprecate and remove support for their plugins (such as [CheckPassword](https://doc.dovecot.org/configuration_manual/authentication/checkpassword/)). DMS will drop support immediately if at some moment continued inclusion of the Lua plugin would not align with the DMS development process anymore.
```

Comment:
I made some slight alterations to the suggestion. It can concern multiple packages, depending on whether community repositories are used. The details of this are not relevant for this community guide, so I made it a bit more generic without taking anything away from the warning.

This will be resolved in the next documentation commit.